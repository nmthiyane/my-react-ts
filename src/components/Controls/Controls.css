div.Controls {
	margin-top: 20px;
	display: flex;
	flex-direction: row;
	justify-content: center;
}

.Controls > button {
	margin: 5px;
	padding: 5px 15px;
	border: 2px solid #014373;
	color: #014373;
	transition: background-color .5s, color .3s;
	cursor: pointer;
}

.Controls > button:hover {
	background-color: #014373;
	color: white;
}

.Controls > button:disabled {
	cursor: not-allowed;
	background-color: #014373;
	color: white;
}

.Controls > button:disabled:hover {
	background-color: #014373;
	color: white;
}