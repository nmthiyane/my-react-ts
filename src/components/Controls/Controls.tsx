import * as React from 'react';
import './Controls.css';

interface Props {
	raceInProgress?: boolean;
	raceStartPosition?: boolean;
	resetRace?: (event: MouseEventInit) => void;
	startRace?: (event: MouseEventInit) => void;
}

export class Controls extends React.Component<Props, {}> {
	constructor(props: Props) {
		super(props);
	}
	render() {
		const {
			raceInProgress,
			startRace,
			raceStartPosition,
			resetRace
		} = this.props;

		return (
			<div className="Controls">
				<button
					disabled={!raceStartPosition || raceInProgress}
					type="button"
					onClick={startRace}
				>
					Start
				</button>
				<button
					disabled={raceStartPosition || raceInProgress}
					type="button"
					onClick={resetRace}
				>
					Reset
				</button>
			</div>
		);
	}
}