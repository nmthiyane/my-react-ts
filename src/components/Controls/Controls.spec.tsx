import * as React  from 'react';
import { Controls } from './Controls';
const renderer  = require('react-test-renderer');

describe('Snapshot test for the Controls component', () => {

	it('renders a snapshot', () => {
	const tree = renderer.create((
		<Controls 
		/>
		)).toJSON();
	expect(tree).toMatchSnapshot();
	});
});