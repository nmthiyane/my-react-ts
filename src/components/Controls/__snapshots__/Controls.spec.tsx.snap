// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the Controls component renders a snapshot 1`] = `
<div
  className="Controls"
>
  <button
    disabled={true}
    onClick={undefined}
    type="button"
  >
    Start
  </button>
  <button
    disabled={undefined}
    onClick={undefined}
    type="button"
  >
    Reset
  </button>
</div>
`;
